registry/remote/credentials/memory_store.go

```diff
@@ -17,23 +17,50 @@ package credentials
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"sync"
 
 	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/credentials/internal/config"
 )
 
-// memoryStore is a store that keeps credentials in memory.
-type memoryStore struct {
+// MemoryStore is a store that keeps credentials in memory.
+type MemoryStore struct {
 	store sync.Map
 }
 
 // NewMemoryStore creates a new in-memory credentials store.
 func NewMemoryStore() Store {
-	return &memoryStore{}
+	return &MemoryStore{}
+}
+
+// NewMemoryStoreFromConfig creates a new in-memory credentials store from the given configuration.
+//
+// Reference: https://docs.docker.com/engine/reference/commandline/cli/#docker-cli-configuration-file-configjson-properties
+func NewMemoryStoreFromConfig(c []byte) (*MemoryStore, error) {
```

Comment:
It is better to call out it is docker config.
```suggestion
func NewMemoryStoreFromDockerConfig(c []byte) (*MemoryStore, error) {
```